bug: Add-vRLIAuthenticationWSA - Configuring Workspace ONE Access Integration in vRealize Log Insight (sfo-vrli01.sfo.rainpole.io) with (sfo-wsa01.sfo.rainpole.io): POST_VALIDATION_FAILED

### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware/power-validated-solutions-for-cloud-foundation/blob/main/CODE_OF_CONDUCT.md).
- [ ] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [ ] Do not leave "+1" or other comments that do not add relevant information or questions.
- [ ] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

5.0

### Module Version

2.6.0.1001

### PowerShell Version

7.3.6

### PowerCLI Version

13.1

### PowerVCF Version

2.3.0.1004

### Guest Operating System

windows server 2019

### Environment Details

Windows 2019+PS7
PT :  5.0.9.1022

### Description

Configuring Workspace ONE Access Integration in vRealize Log Insight (sfo-vrli01.sfo.rainpole.io) with (sfo-wsa01.sfo.rainpole.io): POST_VALIDATION_FAILED

### Error or Debug Output


```powershell

 Version 5.0.9.1022 | Topology: DualRegion-SingleAZ | Networking: Public | Management Domain Tasks: Intelligent Logging and Analytics

 [08-03-2023_12:51:33] [...] [NOTE] Starting the Process of Enabling vRealize Log Insight Integration with Workspace ONE Access
 [08-03-2023_12:51:33] [...] [INFO] Attempting to Enable vRealize Log Insight Integration with Workspace ONE Access Instance (sfo-wsa01.sfo.rainpole.io)
 [08-03-2023_12:51:34] [...] [ERROR] Configuring Workspace ONE Access Integration in vRealize Log Insight (sfo-vrli01.sfo.rainpole.io) with (sfo-wsa01.sfo.rainpole.io): POST_VALIDATION_FAILED
 [08-03-2023_12:51:34] [...] [NOTE] Finished the Process of Enabling vRealize Log Insight Integration with Workspace ONE Access

 Press any key to continue/return to menu...

```
After adding delay, it went fine.
```powershell


    if ((Get-vRLIAuthenticationWSA).enabled -eq $false) {
                                    Set-vRLIAuthenticationWSA -hostname $wsaFqdn -port 443 -redirectUrl $vcfVrliDetails.fqdn -username $wsaUser -password $wsaPass
                                    sleep 2
									Write-Output "After 2sec delay..."								
									if ((Get-vRLIAuthenticationWSA).enabled -eq $true) {
                                        Write-Output "Configuring Workspace ONE Access Integration in vRealize Log Insight ($($vcfVrliDetails.fqdn)) with ($wsaFqdn): SUCCESSFUL"
                                    } else {
                                        Write-Error "Configuring Workspace ONE Access Integration in vRealize Log Insight ($($vcfVrliDetails.fqdn)) with ($wsaFqdn): POST_VALIDATION_FAILED"
                                    }
```

```powershell

 Version 5.0.9.1022 | Topology: DualRegion-SingleAZ | Networking: Public | Management Domain Tasks: Intelligent Logging and Analytics

 [08-03-2023_13:00:12] [...] [NOTE] Starting the Process of Enabling vRealize Log Insight Integration with Workspace ONE Access
 [08-03-2023_13:00:12] [...] [INFO] Attempting to Enable vRealize Log Insight Integration with Workspace ONE Access Instance (sfo-wsa01.sfo.rainpole.io)
 [08-03-2023_13:00:15] [...] [INFO] After 2sec delay... Configuring Workspace ONE Access Integration in vRealize Log Insight (sfo-vrli01.sfo.rainpole.io) with (sfo-wsa01.sfo.rainpole.io): SUCCESSFUL
 [08-03-2023_13:00:15] [...] [NOTE] Finished the Process of Enabling vRealize Log Insight Integration with Workspace ONE Access

 Press any key to continue/return to menu...

```

### Expected Behavior

config should be applied.

### Actual Behavior

Configuring Workspace ONE Access Integration in vRealize Log Insight (sfo-vrli01.sfo.rainpole.io) with (sfo-wsa01.sfo.rainpole.io): POST_VALIDATION_FAILED

### Steps to Reproduce

PT option : 30, 91 13: Workspace One: Enable Authentication for vRealize Log Insight by Using Workspace ONE Access

### Log Fragments and Files

 Version 5.0.9.1022 | Topology: DualRegion-SingleAZ | Networking: Public | Management Domain Tasks: Intelligent Logging and Analytics

 [08-03-2023_12:51:33] [...] [NOTE] Starting the Process of Enabling vRealize Log Insight Integration with Workspace ONE Access
 [08-03-2023_12:51:33] [...] [INFO] Attempting to Enable vRealize Log Insight Integration with Workspace ONE Access Instance (sfo-wsa01.sfo.rainpole.io)
 [08-03-2023_12:51:34] [...] [ERROR] Configuring Workspace ONE Access Integration in vRealize Log Insight (sfo-vrli01.sfo.rainpole.io) with (sfo-wsa01.sfo.rainpole.io): POST_VALIDATION_FAILED
 [08-03-2023_12:51:34] [...] [NOTE] Finished the Process of Enabling vRealize Log Insight Integration with Workspace ONE Access

 Press any key to continue/return to menu...




    if ((Get-vRLIAuthenticationWSA).enabled -eq $false) {
                                    Set-vRLIAuthenticationWSA -hostname $wsaFqdn -port 443 -redirectUrl $vcfVrliDetails.fqdn -username $wsaUser -password $wsaPass
                                    sleep 2
									Write-Output "After 2sec delay..."								
									if ((Get-vRLIAuthenticationWSA).enabled -eq $true) {
                                        Write-Output "Configuring Workspace ONE Access Integration in vRealize Log Insight ($($vcfVrliDetails.fqdn)) with ($wsaFqdn): SUCCESSFUL"
                                    } else {
                                        Write-Error "Configuring Workspace ONE Access Integration in vRealize Log Insight ($($vcfVrliDetails.fqdn)) with ($wsaFqdn): POST_VALIDATION_FAILED"
                                    }


 Version 5.0.9.1022 | Topology: DualRegion-SingleAZ | Networking: Public | Management Domain Tasks: Intelligent Logging and Analytics

 [08-03-2023_13:00:12] [...] [NOTE] Starting the Process of Enabling vRealize Log Insight Integration with Workspace ONE Access
 [08-03-2023_13:00:12] [...] [INFO] Attempting to Enable vRealize Log Insight Integration with Workspace ONE Access Instance (sfo-wsa01.sfo.rainpole.io)
 [08-03-2023_13:00:15] [...] [INFO] After 2sec delay... Configuring Workspace ONE Access Integration in vRealize Log Insight (sfo-vrli01.sfo.rainpole.io) with (sfo-wsa01.sfo.rainpole.io): SUCCESSFUL
 [08-03-2023_13:00:15] [...] [NOTE] Finished the Process of Enabling vRealize Log Insight Integration with Workspace ONE Access

 Press any key to continue/return to menu...







### Screenshots

_No response_

### References

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,9 @@
 
 > Release Date: Unreleased
 
+- Enhanced `Add-vRLISmtpConfiguration` cmdlet with a 2 second delay to ensure the API call completes.
+- Enhanced `Add-vRLIAuthenticationWSA` cmdlet with a 2 second delay to ensure the API call completes.
+- Enhanced `Add-vRLIAgentGroup` cmdlet with a 2 second delay to ensure the API call completes.
 - Added `Get-vRSLCMProductBinariesMapped` cmdlet to retrieves a list of mapped Product Binaries in vRealize Suite Lifecycle Manager.
 - Added `Get-vRSLCMProductBinaries` cmdlet to retrieve a list of product binaries in vRealize Suite Lifecycle Manager.
 - Added `Register-vRSLCMProductBinary` cmdlet to add a product binary to the vRealize Suite Lifecycle Manager inventory.

--- a/PowerValidatedSolutions.psd1
+++ b/PowerValidatedSolutions.psd1
@@ -12,7 +12,7 @@
     RootModule = 'PowerValidatedSolutions.psm1'
     
     # Version number of this module.
-    ModuleVersion = '2.6.0.1001'
+    ModuleVersion = '2.6.0.1002'
     
     # Supported PSEditions
     # CompatiblePSEditions = @()

--- a/PowerValidatedSolutions.psm1
+++ b/PowerValidatedSolutions.psm1
@@ -7584,6 +7584,7 @@ Function Add-vRLISmtpConfiguration {
                             if (Test-Connection -ComputerName $smtpServer -Quiet -Count 1) {
                                 if (!(Get-vRLISmtpConfiguration | Where-Object {$_.server -eq $smtpServer})) {
                                     Set-vRLISmtpConfiguration -smtpServer $smtpServer -port $port -sender $sender -username $smtpUser -password $smtpPass | Out-Null
+                                    Start-Sleep 2
                                     if (Get-vRLISmtpConfiguration | Where-Object {$_.server -eq $smtpServer}) {
                                         Write-Output "Configuring SMTP Server in vRealize Log Insight ($($vcfVrliDetails.fqdn)) with SMTP server ($smtpServer): SUCCESSFUL"
                                     } else {
@@ -7643,6 +7644,7 @@ Function Add-vRLIAuthenticationWSA {
                             if (Test-Connection -ComputerName $wsaFqdn -Quiet -Count 1) {
                                 if ((Get-vRLIAuthenticationWSA).enabled -eq $false) {
                                     Set-vRLIAuthenticationWSA -hostname $wsaFqdn -port 443 -redirectUrl $vcfVrliDetails.fqdn -username $wsaUser -password $wsaPass
+                                    Start-Sleep 2
                                     if ((Get-vRLIAuthenticationWSA).enabled -eq $true) {
                                         Write-Output "Configuring Workspace ONE Access Integration in vRealize Log Insight ($($vcfVrliDetails.fqdn)) with ($wsaFqdn): SUCCESSFUL"
                                     } else {
@@ -7861,6 +7863,7 @@ Function Add-vRLIAgentGroup {
                         if (Test-vRLIAuthentication -server $vcfVrliDetails.fqdn -user $vcfVrliDetails.adminUser -pass $vcfVrliDetails.adminPass) {
                             if (!(Get-vRLIAgentGroup | Select-Object name | Where-Object {$_.name -eq $agentGroupName})) {
                                 New-vRLIAgentGroup -agentGroupType $agentGroupType -criteria $criteria -agentGroupName $agentGroupName | Out-Null
+                                Start-Sleep 2
                                 if (Get-vRLIAgentGroup | Select-Object name | Where-Object {$_.name -eq $agentGroupName}) {
                                     Write-Output "Creating Agent Group in vRealize Log Insight ($($vcfVrliDetails.fqdn)) for ($agentGroupName): SUCCESSFUL"
                                 } else {

